cmd/controller/state/kubelinter/customchecks/bindings/bindings.go

```diff
@@ -0,0 +1,160 @@
+)
+
+func Checks() []*config.Check {
+	var checks []*config.Check
+	for k, v := range rules {
+		checks = append(checks, &config.Check{
+			Name:        k,
+			Template:    templateName,
+			Description: v.Name,
+			Params: map[string]interface{}{
+				"Values":         v.Values,
+				"ExcludedValues": v.ExcludedValues,
+			},
+		})
+	}
+
+	return checks
+}
+
+func init() {
+	templates.Register(roleBindingsTemplate())
+}
+
+var (
+	rules = map[string]BindingsCheck{
+		"system-masters": {
+			Name:   "Avoid use of system:masters group",
+			Values: []string{"system:masters"},
+		},
+		"system-anonymous": {
+			Name:   "Avoid binding to system:anonymous",
+			Values: []string{"system:anonymous"},
+		},
+		"system-unauthenticated": {
+			Name:           "Avoid non-default bindings to system:unauthenticated",
+			Values:         []string{"system:unauthenticated"},
+			ExcludedValues: []string{"system:public-info-viewer"},
+		},
+		"system-authenticated": {
+			Name:           "Avoid non-default bindings to system:authenticated",
+			Values:         []string{"system:authenticated"},
+			ExcludedValues: []string{"system:public-info-viewer", "system:basic-user", "system:discovery"},
+		},
+	}
+)
+
+type BindingsCheck struct {
+	Name           string
+	Values         []string
+	ExcludedValues []string
+}
+
+func roleBindingsTemplate() check.Template {
+	template := check.Template{
+		HumanName: "Avoid bindings to certain groups",
+		Key:       templateName,
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{
+				objectkinds.RoleBinding,
+				objectkinds.ClusterRoleBinding,
+			},
+		},
+		Parameters:             ParamDescs,
+		ParseAndValidateParams: ParseAndValidate,
+		Instantiate: WrapInstantiateFunc(func(p Params) (check.Func, error) {
+			return func(_ lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				var subjects []rbacv1.Subject
+				var roleRef rbacv1.RoleRef
+
+				rb, ok := object.K8sObject.(*rbacv1.RoleBinding)
+				if !ok {
+					crb, ok := object.K8sObject.(*rbacv1.ClusterRoleBinding)
+					if !ok {
+						return nil
+					}
+
+					subjects = crb.Subjects
+					roleRef = crb.RoleRef
+				} else {
+					subjects = rb.Subjects
+					roleRef = rb.RoleRef
+				}
+
+				for _, subject := range subjects {
+					if subject.Kind == "Group" && slices.Contains(p.Values, subject.Name) {
+						if p.ExcludedValues == nil || !slices.Contains(p.ExcludedValues, roleRef.Name) {
+							return []diagnostic.Diagnostic{{Message: fmt.Sprintf("Binding to %s", subject.Name)}}
+						}
+					}
+				}
+
+				return nil
+			}, nil
+		}),
+	}
+
+	return template
+}
+
+type Params struct {
+	Values         []string
+	ExcludedValues []string
+}
+
+var (
+	// Use some imports in case they don't get used otherwise.
+	_ = util.MustParseParameterDesc
+	_ = fmt.Sprintf
+
+	ParamDescs = []check.ParameterDesc{}
+)
+
+func (p *Params) Validate() error {
+	var validationErrors []string
+	if len(validationErrors) > 0 {
```

Comment:
This is always false, because the nil slice is created for each call of this function.